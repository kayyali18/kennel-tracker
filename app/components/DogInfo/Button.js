import styled from 'styled-components'

const Button = styled.div`
  margin: 0 auto;
  text-decoration: none;
  text-align: center;
  color: grey;
  border: none;
  position: relative;
  height: 35px;
  width: 500px;
  font-size: 1.6em;
  padding: 0 0.5em;
  cursor: pointer;
  transition: 800ms ease all;
  outline: none;

  &:before,
  &:after {
    content: '';
    text-decoration: none;
    position: absolute;
    top: 0;
    right: 0;
    height: 2px;
    width: 0;
    background: grey;
    transition: 400ms ease all;
  }

  &:after {
    right: inherit;
    top: inherit;
    text-decoration: none;
    left: 0;
    bottom: 0;
  }
  &:hover {
    text-decoration: none;
    color: grey;
  }

  &:hover:before,
  &:hover:after {
    width: 100%;
    text-decoration: none;
    transition: 800ms ease all;
  }
`
export default Button
