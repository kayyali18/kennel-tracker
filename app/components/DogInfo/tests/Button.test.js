import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Button from '../Button'

describe('<Button />', () => {
  it('should match snapshot', () => {
    const renderedComponent = renderer.create(<Button />).toJSON()
    expect(renderedComponent).toMatchSnapshot()
  })

  it('should render a div', () => {
    const renderedComponent = renderer.create(<Button />).toJSON()
    expect(renderedComponent.type).toEqual('div')
  })
})
