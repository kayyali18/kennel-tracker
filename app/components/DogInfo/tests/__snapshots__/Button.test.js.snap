// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
.c0 {
  margin: 0 auto;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  color: grey;
  border: none;
  position: relative;
  height: 35px;
  width: 500px;
  font-size: 1.6em;
  padding: 0 0.5em;
  cursor: pointer;
  -webkit-transition: 800ms ease all;
  transition: 800ms ease all;
  outline: none;
}

.c0:before,
.c0:after {
  content: '';
  -webkit-text-decoration: none;
  text-decoration: none;
  position: absolute;
  top: 0;
  right: 0;
  height: 2px;
  width: 0;
  background: grey;
  -webkit-transition: 400ms ease all;
  transition: 400ms ease all;
}

.c0:after {
  right: inherit;
  top: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  left: 0;
  bottom: 0;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: grey;
}

.c0:hover:before,
.c0:hover:after {
  width: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 800ms ease all;
  transition: 800ms ease all;
}

<div
  className="c0"
/>
`;
